fiftyone/core/stages.py

```
@@ -3635,16 +3646,21 @@ def _make_flat_pipeline(self, sample_collection):
                 {"$unwind": "$docs"},
                 {"$replaceRoot": {"newRoot": "$docs"}},
-                {"$addFields": {"_group": group_expr}},
             ]
         )
 
+        if self._include_group:
+            pipeline.append(self._group_field_stage(sample_collection))
+
         return pipeline
 
     def _make_grouped_pipeline(self, sample_collection):
```

Ruff (0.15.20)

[warning] Missing return type annotation for private function `_make_grouped_pipeline`

(ANN202)